capstone-server/src/main/java/com/hcl/capstoneserver/config/GlobalErrorHandler.java

```diff
@@ -85,5 +90,32 @@
                 HttpStatus.UNAUTHORIZED
         );
     }
+
+    @ExceptionHandler(EmailAlreadyExistsException.class)
+    public final ResponseEntity<Object> handleEmailExistsException(EmailAlreadyExistsException ex) {
+        ArrayList<Map<String, String>> errors = errorsMap("email", ex.getMessage());
+        return new ResponseEntity<>(
+                new DefaultValidationErrorResponse(HttpStatus.BAD_REQUEST, "User Already Exists", errors),
+                HttpStatus.BAD_REQUEST
+        );
+    }
+
+    @ExceptionHandler(UserAlreadyExistsException.class)
+    protected final ResponseEntity<Object> handleUserIdExistsException(UserAlreadyExistsException ex) {
+        ArrayList<Map<String, String>> errors = errorsMap("userId", ex.getMessage());
+        return new ResponseEntity<>(
+                new DefaultValidationErrorResponse(HttpStatus.BAD_REQUEST, "User Already Exists", errors),
+                HttpStatus.BAD_REQUEST
+        );
+    }
+
+    private ArrayList<Map<String, String>> errorsMap(String field, String message) {
```

Comment:
if we are going to use a general method to create the error map, it should support generating multiple errors like the first method